.github/workflows/e2e-tests.yml

```diff
@@ -44,171 +44,220 @@ jobs:
-
-        # Update permissions to workaround https://github.com/actions/cache/issues/753
-        sudo chown -R runner:docker "$GITHUB_WORKSPACE/tools/docker"
-
-    - name: Upload test artifacts
-      if: ${{ always() }}
-      continue-on-error: true
-      uses: actions/upload-artifact@v3
-      with:
-        name: test-output-${{ matrix.project }}
-        path: ${{ matrix.path }}/output
+          fi
+
+          if [ "${SUITE}" == gutenberg ]; then
+              echo "::group::Setting up Gutenberg"
+              pnpm e2e-env gb-setup
+              echo "::endgroup::"
+          fi
+
+      - name: Wait for atomic site to be ready
+        if: ${{ startsWith(matrix.suite, 'atomic') }}
+        working-directory: ${{ matrix.path }}
+        env:
+          DISPATCH_REF_NAME: ${{ github.event.client_payload.ref_name }}
+          DISPATCH_REF_TYPE: ${{ github.event.client_payload.ref_type }}
+        run: |
+          SITE="atomic"
+
+          if [[ "${DISPATCH_REF_NAME}" != trunk ]]; then
+              SITE="atomicRc"
+          fi
+
+          echo "TEST_SITE=${SITE}" >> $GITHUB_ENV
+
+          # Wait for the site to pick up latest tag version (DISPATCH_REF_NAME)
+          TEST_SITE="${SITE}" node "$GITHUB_WORKSPACE/tools/e2e-commons/bin/update-beta-version.cjs" $DISPATCH_REF_TYPE $DISPATCH_REF_NAME
+
+      - name: Set up VIP site
+        if: ${{ matrix.suite == 'vip' }}
+        working-directory: ${{ matrix.path }}
+        run: |
+          echo "TEST_SITE=vip" >> $GITHUB_ENV
+
+      - name: Run ${{ matrix.project }} tests
+        working-directory: ${{ matrix.path }}
+        env:
+          JSON_ARGS: ${{ toJSON(matrix.testArgs) }}
+        run: |
+          mapfile -t TEST_ARGS < <(jq -r '.[]' <<<"$JSON_ARGS")
+          pnpm run test:run "${TEST_ARGS[@]}"
+
+      - name: Write report metadata
+        if: ${{ always() }}
+        env:
+          SUITE: ${{ matrix.suite }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          OUTPUT_PATH: ${{ matrix.path }}
+        run: .github/files/e2e-tests/report-metadata.sh
+
+      - name: Test environment tear-down
+        if: ${{ always() }}
+        working-directory: ${{ matrix.path }}
+        continue-on-error: true
+        run: |
+          pnpm run tunnel:down
+          echo "::group::Tunnel logs"
+          cat "$HOME/tunnel.log"
+          echo "::endgroup::"
+
+          # Update permissions to workaround https://github.com/actions/cache/issues/753
+          sudo chown -R runner:docker "$GITHUB_WORKSPACE/tools/docker"
+
+      - name: Upload test artifacts
+        if: ${{ always() }}
+        continue-on-error: true
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-output-${{ matrix.project }}
+          path: ${{ matrix.path }}/output
 
   test-report:
     name: "Test report"
     runs-on: ubuntu-latest
-    if: ${{ success() || failure() }}
-    needs: [e2e-tests]
+    if: ${{ always() }}
```

Comment:
Using `success() || failure()` rather than `always()` here was intentional, see https://github.com/Automattic/jetpack/pull/28913#discussion_r1105079565

Reply:
They seemed intentional (although might be worth linking to that comment via a comment), but I assumed that this might run into the same problem as I mentioned earlier - the job not even trying to start if the previous job in the chain was cancelled.

Reply:
> but I assumed that this might run into the same problem as I mentioned earlier - the job not even trying to start if the previous job in the chain was cancelled.

That appears to be intentional too. The following comment after the one I linked shows some tests with cancelled runs which did not run these jobs and that was described as "validated".

Another good reason to fix the bogus "cancelled" you introduced above (see line 58).